piper/ratbagd.py

```diff
@@ -790,20 +789,23 @@ def cancel(self):
         self._macro = deepcopy(self._old_macro)
         self.notify("keys")
 
-    def append(self, type, keycode):
+    def append(self, type, value):
         """Appends the given event to the current macro.
 
         @param type The type of event, as one of RatbagdButton.MACRO_*.
-        @param keycode The keycode of the event, as int in X.Org or Gdk value.
+        @param value If the type denotes a key event, the X.Org or Gdk keycode
+                     of the event, as int. Otherwise, the value of the timeout
+                     in milliseconds, as int.
         """
-        if not self._XORG_KEYCODE_OFFSET <= value <= 255:
-            print("Keycode is not within the valid range.", file=sys.stderr)
-            return
-        keycode = keycode - self._XORG_KEYCODE_OFFSET
+        if type != RatbagdButton.MACRO_WAIT:
+            if not self._XORG_KEYCODE_OFFSET <= value <= 255:
+                print("Keycode is not within the valid range.", file=sys.stderr)
+                return
+            value = value - self._XORG_KEYCODE_OFFSET
         # Only append if the entry isn't identical to the last one, as we cannot
         # e.g. have two identical key presses in a row.
-        if len(self._macro) == 0 or (type, keycode) != self._macro[-1]:
-            self._macro.append((type, keycode))
+        if len(self._macro) == 0 or (type, value) != self._macro[-1]:
+            self._macro.append((type, value))
             self.notify("keys")
```

Comment:
After moving the keycode handling up to the caller this commit indeed doesn't make sense anymore.